src/Native/Unix/System.Native/pal_io.cpp

```diff
@@ -1171,3 +1171,44 @@ extern "C" char* SystemNative_RealPath(const char* path)
     assert(path != nullptr);
     return realpath(path, nullptr);
 }
+
+static int32_t LockUnlock(intptr_t fd, off_t offset, off_t length, bool lock)
```

Comment:
Rather than a bool, the caller could just pass in F_WRLCK or F_UNLCK, avoiding the branch, and potentially even making the call sites more understandable.